ibis/backends/flink/utils.py

```diff
@@ -282,6 +282,17 @@ def translate_literal(op: ops.Literal) -> str:
             raise ValueError("NaN is not supported in Flink SQL")
         elif math.isinf(value):
             raise ValueError("Infinity is not supported in Flink SQL")
+        elif dtype.is_decimal():
+            # When PyFlink infers schema from `decimal.Decimal` objects,
+            # it will be `DecimalType(38, 18)`.
+            # https://github.com/apache/flink/blob/release-1.17.1/flink-python/pyflink/table/types.py#L336-L337
+            precision = 38 if dtype.precision is None else dtype.precision
+            scale = 18 if dtype.scale is None else dtype.scale
+
+            if precision > 38:
+                raise ValueError("The precision can be up to 38 in Flink")
```

Comment:
I don't see other backends/tests do this; however, if I don't raise an error early, the resulting error on the PyFlink side is something uninformative about writing a NULL to a non-nullable field. I assume this happens because, with invalid input, it makes the value NULL. Is this way to handle it OK?